flask_monitoringdashboard/database/request.py

```diff
@@ -2,12 +2,29 @@
 Contains all functions that access a Request object.
 """
 import time
+from random import sample
 
 from sqlalchemy import and_
 
 from flask_monitoringdashboard.database import Request
 
 
+def get_latencies_in_timeframe(db_session, endpoint_id, start_date, end_date):
+    criterion = create_time_based_sample_criterion(start_date, end_date)
+
+    items = db_session.query(Request.duration).filter(and_(Request.endpoint_id == endpoint_id, *criterion)).all()
+
+    return [item.duration for item in items]
+
+
+def get_latencies_sample(db_session, endpoint_id, interval, sample_size=500):
+    latencies = get_latencies_in_timeframe(db_session, endpoint_id, interval.start_date(), interval.end_date())
+
+    # todo: sample in the database, not in the application
```

Comment:
Done. Bit more difficult than I thought it would be. There's no database independent way of doing it.  For sqlite and mysql I made it sample "in-database", for other DBs it still happens in memory.